code/modules/reagents/withdrawal/generic_addiction.dm

```diff
@@ -0,0 +1,256 @@
+
+/datum/addiction/maintenance_drugs
+	name = "maintenance drug"
+	withdrawal_stage_messages = list("", "", "")
+
+/datum/addiction/maintenance_drugs/withdrawal_enters_stage_1(mob/living/carbon/affected_carbon)
+	. = ..()
+	affected_carbon.hal_screwyhud = SCREWYHUD_HEALTHY
+
+/datum/addiction/maintenance_drugs/withdrawal_stage_1_process(mob/living/carbon/affected_carbon, delta_time)
+	. = ..()
+	if(DT_PROB(7.5, delta_time))
+		affected_carbon.emote("growls")
+
+/datum/addiction/maintenance_drugs/withdrawal_enters_stage_2(mob/living/carbon/affected_carbon)
+	. = ..()
+	if(!ishuman(affected_carbon))
+		return
+	var/mob/living/carbon/human/affected_human = affected_carbon
+	var/obj/item/organ/tongue/T = affected_human.getorganslot(ORGAN_SLOT_TONGUE)
+	if(affected_human.gender == MALE)
+		to_chat(affected_human, "<span class='warning'>Your chin itches.</span>")
+		affected_human.update_hair()
+	//Only like gross food
+	T?.liked_food = GROSS
+	T?.disliked_food = NONE
+	T?.toxic_food = ~GROSS
+
+/datum/addiction/maintenance_drugs/withdrawal_enters_stage_3(mob/living/carbon/affected_carbon)
+	. = ..()
+	if(!ishuman(affected_carbon))
+		return
+	to_chat(affected_carbon, "<span class='warning'>You feel yourself adapt to the darkness.</span>")
+	var/mob/living/carbon/human/affected_human = affected_carbon
+	var/obj/item/organ/eyes/empowered_eyes = affected_human.getorgan(/obj/item/organ/eyes)
+	if(empowered_eyes)
+		ADD_TRAIT(affected_human, TRAIT_NIGHT_VISION, "maint_drug_addiction")
+		empowered_eyes?.refresh()
+
+/datum/addiction/maintenance_drugs/withdrawal_stage_3_process(mob/living/carbon/affected_carbon, delta_time)
+	if(!ishuman(affected_carbon))
+		return
+	var/mob/living/carbon/human/affected_human = affected_carbon
+	var/turf/T = get_turf(affected_human)
+	var/lums = T.get_lumcount()
+	if(lums > 0.5)
+		SEND_SIGNAL(affected_human, COMSIG_ADD_MOOD_EVENT, "too_bright", /datum/mood_event/bright_light)
+		affected_human.dizziness = min(40, affected_human.dizziness + 3)
+		affected_human.confused = min(affected_human.confused, 20)
+	else
+		SEND_SIGNAL(affected_carbon, COMSIG_CLEAR_MOOD_EVENT, "too_bright")
+
+/datum/addiction/maintenance_drugs/end_withdrawal(mob/living/carbon/affected_carbon)
+	. = ..()
+	affected_carbon.hal_screwyhud = SCREWYHUD_NONE
+	if(!ishuman(affected_carbon))
+		return
+	var/mob/living/carbon/human/affected_human = affected_carbon
+	var/obj/item/organ/tongue/T = affected_human.getorganslot(ORGAN_SLOT_TONGUE)
+	T?.liked_food = initial(T?.liked_food)
+	T?.disliked_food = initial(T?.disliked_food)
+	T?.toxic_food = initial(T?.toxic_food)
+	REMOVE_TRAIT(affected_human, TRAIT_NIGHT_VISION, "maint_drug_addiction")
+	var/obj/item/organ/eyes/eyes = affected_human.getorgan(/obj/item/organ/eyes)
+	eyes.refresh()
+
+///Makes you a hypochondriac - I'd like to call it hypochondria, but "I could use some hypochondria" doesn't work
+/datum/addiction/medicine
+	name = "medicine"
+	withdrawal_stage_messages = list("", "", "")
```

Comment:
some messages would be nice,
"I think I might be sick...", "Oh god, I'm dying!", "I need a doctor to check me out, right now!!"

Reply:
saying a "doctor" would be a bit on the nose.

I will try to convey panic though

Reply:
aba9a1c

done